Civi/Core/LocalizationInitializer.php

```diff
@@ -0,0 +1,160 @@
+      $settings = json_decode($json, TRUE);
+
+      if (!empty($settings)) {
+        // get all valid settings
+        $results = civicrm_api3('Setting', 'getfields', array());
+        $validSettings = array_keys($results['values']);
+        // add valid settings to params to send to api
+        foreach ($settings as $setting => $value) {
+          if (in_array($setting, $validSettings)) {
+            $settingsParams[$setting] = $value;
+          }
+
+        }
+
+        // ensure we don't mess with multilingual
+        unset($settingsParams['languageLimit']);
+
+        // support for enabled languages (option group)
+        if (isset($settings['languagesOption']) && count($settings['languagesOption']) > 0) {
+          self::updateLanguages($settings['languagesOption']);
+        }
+
+        // set default currency in currencies_enabled (option group)
+        if (isset($settings['defaultCurrency'])) {
+          self::updateCurrencies(array($settings['defaultCurrency']), $settings['defaultCurrency']);
+        }
+
+      }
+
+    }
+
+    // in any case, enforce the seedLanguage as the default language
+    $settingsParams['lcMessages'] = $seedLanguage;
+
+    // apply the config
+    civicrm_api3('Setting', 'create', $settingsParams);
+
+  }
+
+  /**
+   * Enable provided languages and disable all others
+   *
+   * @param $languages array of languages (locale code) ['fr_CA', 'en_CA']
+   * @throws \CiviCRM_API3_Exception
+   */
+  protected static function updateLanguages($languages) {
+    $result = civicrm_api3('OptionValue', 'get', array(
+      'option_group_id' => "languages",
+      'options' => array('limit' => 0),
+    ));
+
+    foreach ($result['values'] as $id => $language) {
+      $params = $language;
+      if (in_array($language['name'], $languages)) {
+        $params['is_active'] = 1;
+      }
+      else {
+        // disable every language that is not in the given list
+        $params['is_active'] = 0;
+      }
+      civicrm_api3('OptionValue', 'create', $params);
+    }
+
+  }
+
+  /**
+   * Replace available currencies by the ones provided
+   *
+   * @param $currencies array of currencies ['USD', 'CAD']
+   * @param $default default currency
+   */
+  protected static function updateCurrencies($currencies, $default) {
+
+    // sort so that when we display drop down, weights have right value
+    sort($currencies);
+
+    // get labels for all the currencies
+    $options = array();
+
+    $currencySymbols = \CRM_Admin_Form_Setting_Localization::getCurrencySymbols();
+    for ($i = 0; $i < count($currencies); $i++) {
+      $options[] = array(
+        'label' => $currencySymbols[$currencies[$i]],
+        'value' => $currencies[$i],
+        'weight' => $i + 1,
+        'is_active' => 1,
+        'is_default' => $currencies[$i] == $default,
+      );
+    }
+
+    $dontCare = NULL;
+    \CRM_Core_OptionGroup::createAssoc('currencies_enabled', $options, $dontCare);
```

Comment:
I don't think so. The createAssoc function definition is `public static function createAssoc($groupName, &$values, &$defaultID, $groupTitle = NULL) `
I might be wrong but i think i have to pass a variable as it is passed as reference.